sw/device/tests/aon_timer_wakeup_irq_test.c

```diff
@@ -0,0 +1,126 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/freestanding/limits.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+static const uint32_t kAonTimerFreq = 200 * 1000;  // 200kHz
+static const uint32_t kSleepTime_us = 2 * 1000;    // 2ms
+static dif_aon_timer_t aon_timer;
+static dif_rv_plic_t plic;
+static dif_rv_plic_irq_id_t plic_irq_id;
+static ibex_timeout_t timeout;
+
+static void aon_timer_wakeup_config(dif_aon_timer_t *aon_timer,
+                                    uint32_t wakeup_threshold) {
+  // Make sure that wake-up timer is stopped.
+  CHECK_DIF_OK(dif_aon_timer_wakeup_stop(aon_timer));
+
+  // Make sure the wake-up IRQ is cleared to avoid false positive.
+  CHECK_DIF_OK(dif_aon_timer_irq_acknowledge(aon_timer,
+                                             kDifAonTimerIrqWkupTimerExpired));
+
+  // Check whether the irq is pending
+  bool is_pending;
```

Comment:
I suggest initialising `is_pending` to a value that will fail the test if it is left unchanged (just in case `dif_aon_timer_irq_is_pending` doesn't write to it for some reason).